test/plugins/metrics.test.js

```diff
@@ -63,7 +62,11 @@ describe('request metrics plugin', function() {
                     assert.isAtLeast(metrics.latency, 150);
                     assert.isAtLeast(metrics.totalLatency, 150);
                     assert.equal(metrics.path, '/foo');
-                    assert.equal(metrics.connectionState, undefined);
+                    // close doesn't always fire in Node < 10
+                    assert.include(
+                        [undefined, 'close'],
```

Comment:
The problem with using `assert.include` here is that if we run this test with Node.js >= 10.x and we expect the connection state to be `'close'`, the test would pass even if the connection state was `undefined`. Would it be too much work to differentiate between node >= 10.x and < 10.x here and have two different asserts, something like:

```
if (process.version >= 10.0) {
  assert.equal(metrics.connectionState, 'close');
} else {
  assert.equal(metrics.connectionState, undefined);
}
```
?

Same for all other connection state assertions in this file.